tools/tasks/webpack-dev-server.js

```diff
@@ -37,18 +36,34 @@ module.exports = function startWebpackDevServer(webpackConfig, devServerConfig)
     ];
 
     // Make a new server and store a reference to it so we can interact with it later
-    wpds = new WebpackDevServer(webpack(localWebpackConfig), localDevServerConfig);
+    const wpds = new WebpackDevServer(webpack(localWebpackConfig), localDevServerConfig);
     wpds.disableHostCheck = true;
 
     wpds.listen(port, host, (err) => {
       if (err) {
         cb(err);
-        return false;
       }
 
       console.info(`Listening at ${localHost}`);
       cb();
-      return true;
     });
+
+    return wpds;
   };
+}
+
+/**
+ * Take an instance of webpack dev server, send it random state. Causes reload.
+ * @param server
+ * @param cb
+ */
+function refreshWebpackDevServer(server, cb) {
```

Comment:
I've read through this, and from the changeset alone I still don't understand the concept. Maybe it needs more comments, or maybe I'm personally lacking context. Is the idea that that you finish all compilation then send some random querystring at the server to indicate a change instead of using a pure file watch?

Reply:
@grayside That was the understanding I got from the little walkthrough bloom did this afternoon.

Reply:
Docker won't play nice with us watching compiled file output, nor do we really want to watch that many files just for an indicator of "yep, time to reload the page".

So our change is this:

1. We instantiate a little webpack dev server
1. We return that server somewhere that can hold on to it
1. We can refresh it manually again any time we want

We refresh it manually after the pattern lab command line fully returns successful (rather than watching the pl compiled output directory for files changing). We do this in the gulp task named `webpack:watch:pl-source`.

Reply:
And a quick note about the random string: webpack dev server determines whether it needs to do a "hard reload" or "soft reload" based on comparing current state hash to new state hash. We just send over junk and webpack dev server goes "oh snap, hard reload time!". This is only needed for our HTML files generated by PL since they are not on the dependency chain of webpack.

Reply:
Could we convert some of these comments to inline comments?